ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -0,0 +1,199 @@
+   * @return the used capacity of the {@link Store}
+   */
+  public long getUsedCapacity() {
+    return log.getUsedCapacity();
+  }
+
+  /**
+   * Gets the total capacity of the {@link Store}.
+   * @return the total capacity of the {@link Store}.
+   */
+  public long getTotalCapacity() {
+    return capacityInBytes;
+  }
+
+  /**
+   * Gets the size of valid data of all log segments
+   @return a {@link Pair} of reference time at which stats was collected to a {@link SortedMap} of log segment name
+   to the respective valid data size
+   */
+  Pair<Long, SortedMap<String, Long>> getValidDataSizeBySegment() throws IOException, StoreException {
+    SortedMap<String, Long> validDataSize = new TreeMap<>();
+    Pair<Long, Long> validDataSizeBySegment = getValidDataSize(validDataSize);
+    return new Pair<>(validDataSizeBySegment.getFirst(), validDataSize);
+  }
+
+  /**
+   * Gets the used capacity of each log segment.
+   * @return a {@link Pair} of reference time at which stats was collected to a {@link SortedMap} of log segment name
+  to the respective used capacity
+   */
+  Pair<Long, SortedMap<String, Long>> getUsedCapacityBySegment() {
+    long referenceTime = time.milliseconds();
+    SortedMap<String, Long> usedCapacityBySegments = new TreeMap<>();
+    Iterator<IndexSegment> indexSegmentIterator = index.indexes.values().iterator();
+    IndexSegment indexSegment = null;
+    if (indexSegmentIterator.hasNext()) {
+      indexSegment = indexSegmentIterator.next();
+    }
+    while (indexSegmentIterator.hasNext()) {
+      IndexSegment nextIndexSegment = indexSegmentIterator.next();
+      if (!nextIndexSegment.getLogSegmentName().equals(indexSegment.getLogSegmentName())) {
+        // index segment refers to a new log segment compared to previous index segment
+        String logSegmentName = indexSegment.getLogSegmentName();
+        usedCapacityBySegments.put(logSegmentName, log.getSegment(logSegmentName).getEndOffset());
+        indexSegment = nextIndexSegment;
+      }
+    }
+    if (index.indexes.size() > 1) {
+      String logSegmentName = indexSegment.getLogSegmentName();
+      usedCapacityBySegments.put(logSegmentName, log.getSegment(logSegmentName).getEndOffset());
+    }
+    return new Pair<>(referenceTime, usedCapacityBySegments);
+  }
+
+  /**
+   * Derives valid data size for the store
+   * @param validDataSizeBySegments the {@link SortedMap} that needs to be updated with valid data size value
+   *                                for every segment. Could be {@code null} if interested only in total valid data size
+   * @return a {@link Pair} of reference time and total valid data size
+   * @throws IOException
+   * @throws StoreException
+   */
+  private Pair<Long, Long> getValidDataSize(SortedMap<String, Long> validDataSizeBySegments)
+      throws IOException, StoreException {
+    long validDataSize = 0;
+    long totalValidDataSize = 0;
+    long referenceTime = time.milliseconds();
```

Comment:
This is to ensure that all blobs are checked for expiration exactly at the same time. This is much needed for compaction purposes.